python/pyiceberg/expressions/__init__.py

```diff
@@ -126,10 +164,13 @@ def bind(self, schema: Schema, case_sensitive: bool = True) -> BoundReference[T]
         """
         field = schema.find_field(name_or_id=self.name, case_sensitive=case_sensitive)
         accessor = schema.accessor_for_field(field.field_id)
-        return BoundReference(field=field, accessor=accessor)
+        return self.as_bound(field=field, accessor=accessor)
+
+    @property
+    def as_bound(self) -> Type[BoundReference]:
```

Comment:
Why was this working before but needs to be changed now?

Reply:
For the projection transforms we need to have it the other way around `as_unbound`: https://github.com/apache/iceberg/pull/6128/files#diff-6e61823ae852914cf80ce42354961efd01d99b58fa7804d931ba9cf7ba8edac8R312-R315

We can't do it the same way because it creates a circular reference. I think it is nice to have it the same way.